### PR TITLE
small stuff, tests, doc, docgen.py bugfix, doc, small code cleanup

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: theano/sandbox/cuda/tests/test_type.py [hunk 1]
```diff
@@ -8,8 +9,10 @@
 from theano.misc.pkl_utils import CompatUnpickler
 from theano.sandbox.cuda import cuda_available
 
-if cuda_available:
-    from theano.sandbox.cuda import CudaNdarray
+if not cuda_available:
+    raise SkipTest('Optional package cuda disabled')
+
+from theano.sandbox.cuda import CudaNdarray
```
Comment (abergeron) [inline theano/sandbox/cuda/tests/test_type.py]: This does not respect flake8 so either tag it noqa or change it back.

**Comment by abergeron** [comment]
Apart from the flake8 problem I'm ok

**Comment by nouiz** [comment]
I fixed the flake8 error.

**Comment by abergeron** [comment]
Ok, waiting for travis.

**Pull request merged**